### ISSUE TITLE
Initialize RampModel with a raw science file.

### ISSUE THREAD

**Issue by stscijgbot-rstdms** [spacetelescope/romancal#287]
_Issue [RCAL-196](https://jira.stsci.edu/browse/RCAL-196) was created on JIRA by [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva):_

dq_init needs a more convenient way to create a RampModel from a raw science file.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Erica Kolatch](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ekolatch) on [JIRA](https://jira.stsci.edu/browse/RCAL-196?focusedCommentId=546878#comment-546878):_

[Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva) - is there a parent for this ticket?

**Comment by perrygreenfield** [comment]
Is this the issue Paul raised about not being able to read in a science raw file using the RampModel constructor?

**Comment by nden** [comment]
@perrygreenfield Yes, this is it.

**Comment by perrygreenfield** [comment]
In talking with Paul we decided that this can be deferred.  If there are only a few instances of this kind of usage, we just do the conversion explicitly in the step code. I had written some code to handle this, but removed it in my last PR. This may never be needed.

**Comment by sosey** [comment]
what happens when I want to open a level 1 guide window file into a RampModel?

**Comment by perrygreenfield** [comment]
That depends on what you want to do with the data and how different it is from raw science data. I don't know either so pointing me towards something that answers that would help (or a simple summary).

**Comment by sosey** [comment]
When you talk about the "raw" data, you mean the level 1 files, that would be ready to go into level 2 processing, yes? We are only processing guide windows to level 1, they are subarrays on the WFI chip, so the same as "raw" science data. We'll have ints stored on disk that need to be opened to floats in a RampModel so that folks can analyze them as needed and do any following processing that they write themselves. I haven't kept up recently with the work between roman_datamodels and romancal so I could be out of date on this end of things as well, are we still talking about the same things?

**Comment by perrygreenfield** [comment]
By raw, I mean files that haven't had any romancal processing applied (they are constructed by SDF, not us). So no calibration at all.

**Comment by sosey** [comment]
right, the level 1 files that would be ready for level 2 processing, we have the same definition :)

**Comment by perrygreenfield** [comment]
I guess I wonder why they wouldn't open it as a raw file and access the data directly as ints (easy to convert them to floats). Are there tools that they are using that depend on it being a RampModel? (I'm guessing not yet :-)

**Comment by sosey** [comment]
Because we want the dq and error arrays to be created implicitly when they open it? We should be able to perform dq_init on the open model so that badpixels are evident.

**Comment by perrygreenfield** [comment]
OK, invoking dq_init is somewhat different. And in fact, it is that very step that raised this issue. So that step currently generates a ramp model from the raw data. So won't running that step solve the issue?
